Alias strings for AOT backends, and No-decomps for AOTAutograd accuracy test

anijain2305 opened the pull request:
@Chillee suggested a better UX for using AOT Autograd memory efficient fusion. Suggestion is for `BACKENDS['nvfuser]` to point to `aot_autograd_efficient_strategy`. This way, the user does not have to add this extra import from torchdynamo.

Asking for comments @jansel @ngimel if `nvfuser` is a fine replacement. One other possibility is to use `aot_nvfuser` instead, which is little more explicit about using aot.

jansel commented:
Agree we should add an alias string.  `aot_nvfuser` works for me, but fine with either.

anijain2305 commented:
Changed to `aot_nvfuser`

Chillee reviewed (approved):
LGTM, although is the "aot_ts" backend even using NNC now?

anijain2305 commented:
> LGTM, although is the "aot_ts" backend even using NNC now?

Yeah, I realized that even though we were using the word NNC before, it was not really NNC. I just removed the NNC word and replaced it with `ts` to be more clear.

jansel commented:
> LGTM, although is the "aot_ts" backend even using NNC now?

I think so, because of:
https://github.com/pytorch/torchdynamo/blob/f6fd47b2adb833d5b86a5f8e38017b369a537c34/benchmarks/common.py#L1001

Though we should change `args.nvfuser` to default to True at this point.

anijain2305 commented on benchmarks/common.py:
```diff
@@ -698,7 +695,7 @@ def run_one_model(
 
             try:
-                with optimize_ctx:
+                with accuracy_ctx:
```
Just to emphasize - Using a different ctx for accuracy testing. It is same as `optimize_ctx` unless overridden while handling cmdline args.

For aot_autograd, we choose no-decomps

anijain2305 commented:
@jansel @Chillee Can I request one more review please? I am doing a couple of things in this PR

* Adding alias strings for AOT backend
* Switching off Decomps for AOT Autograd testing.

Chillee reviewed (approved):
lgtm.

Chillee commented:
Some documentation about what each "string backend" does would be nice though.
